client_linux_integration_test.go

```diff
@@ -43,4 +43,28 @@ func TestLinuxClientIntegration(t *testing.T) {
 			t.Logf("port: %+v", p)
 		}
 	})
+
+	t.Run("dpipe_tables", func(t *testing.T) {
+		var tables []*devlink.DpipeTable
+		devices, err := c.Devices()
+		if err != nil {
+			t.Fatalf("failed to get devices: %v", err)
+		}
+
+		for _, d := range devices {
+			tt, err := c.DpipeTables(d)
+			if err != nil {
+				t.Errorf("failed to get DPIPE table from device %v: %v", d, err)
+			}
+			tables = append(tables, tt...)
+		}
+
+		if len(tables) == 0 {
```

Comment:
I think it's okay to remove this check and just print the tables that were found. These "integration" tests are more so just a sanity check of the actual devlink interface at the moment. If there is a simulator kernel module of some sort we could use in CI though, that would be very helpful. Do you know of anything like that?

Reply:
Yes, there is a **netdevsim** module available, but it doesn't support DPIPE yet.
We can probably use it to test some other devlink features like devlink trap, for example.
More details on netdevsim and devlink trap:
https://cateee.net/lkddb/web-lkddb/NETDEVSIM.html
https://github.com/torvalds/linux/blob/master/Documentation/networking/devlink-trap.rst
https://github.com/jeffmerkey/leaf-linux/blob/master/tools/testing/selftests/drivers/net/netdevsim/devlink_trap.sh

Reply:
Awesome! I'll open an issue to look into this.